tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/service/impl/PostServiceImpl.java

```diff
@@ -664,7 +669,14 @@ public PostDTO findOne(Long id) {
   @Override
   public void delete(Long id) {
     log.debug("Request to delete Post : {}", id);
-    postRepository.deleteById(id);
+    List<Placement> attachedPlacements = placementRepository.findByPostId(id);
+    if (attachedPlacements.isEmpty()) {
+      postRepository.deleteById(id);
+      List<PostFunding> postFundingsToDelete = postFundingRepository.findByPostId(id);
+      postFundingRepository.deleteAll(postFundingsToDelete);
```

Comment:
Getting the post would mean that JPA joined placements & funding can be fetched and/or deleted without having to do that ourselves and including another dependency (code smell).  Please which is preferable.